Source/Core/DolphinQt2/Resources.cpp

```diff
@@ -2,46 +2,74 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
+#include <QGuiApplication>
+#include <QIcon>
+#include <QScreen>
 #include <QStringList>
 
 #include "Common/CommonPaths.h"
 #include "Common/FileUtil.h"
 #include "DolphinQt2/Resources.h"
+#include "DolphinQt2/Settings.h"
 
 QList<QPixmap> Resources::m_platforms;
 QList<QPixmap> Resources::m_countries;
 QList<QPixmap> Resources::m_ratings;
 QList<QPixmap> Resources::m_misc;
 
+QString Resources::GetPreferredPixmapScale(int max)
+{
+  const auto dpr = QGuiApplication::primaryScreen()->devicePixelRatio();
+  const auto scale = std::min(dpr <= 1 ? 1 : (dpr <= 2) ? 2 : 4, max);
+
+  if (scale > 1)
+    return QStringLiteral("@%1x").arg(scale);
+  else
+    return QStringLiteral("");
+}
+
+QPixmap Resources::GetScaledPixmap(const std::string& name, bool is_themed)
```

Comment:
Boolean parameters that aren't immediately obvious aren't a very good pattern, since it reduces readability and leads to mistakes. One way to deal with them is to introduce a typed enum, but that's overkill here. I'd like to see something like this (which you basically had before, just with the additional dir parameter that shouldn't be exposed):

```c++
QPixmap GetResourcePixmap(const std::string& name);
QIcon GetThemeIcon(const std::string& name);
```

with any functions that take a file/directory path or boolean parameter being static to this file.